tools/gulp/packaging/build-config.ts

```diff
@@ -0,0 +1,17 @@
+/** Interface for the build config. */
+export interface BuildConfig {
+  projectVersion: string;
+  projectDir: string;
+  packagesDir: string;
+  outputDir: string;
+  licenseBanner: string;
+}
+
+/** Build config being used to build package output. */
+export let buildConfig: BuildConfig;
+
+/** Configures the build by setting the build config. */
+export function configureBuild(config: BuildConfig) {
```

Comment:
There are a couple other ways we could approach this. 
1. Each method takes in `buildConfig` as an argument
2. We create a class like `LibraryBuilder` like `new LibraryBuilder(config)` and then do, e.g., `libraryBuilder.composeRelease(config)`. This could delegate to the existing functions, passing along the necessary settings.